src/PersistentCollection/PersistentCollectionTrait.php

```diff
@@ -563,36 +584,37 @@ public function offsetSet($offset, $value)
         $this->doSet($offset, $value, true);
     }
 
-    /**
-     * @param mixed $offset
-     *
-     * @return void
-     */
+    /** @return void */
     #[ReturnTypeWillChange]
-    public function offsetUnset($offset)
+    public function offsetUnset(mixed $offset)
```

Comment:
```suggestion
    public function offsetUnset(mixed $offset): void
```

This also allows to remove the PHP doc and `#[ReturnTypeWillChange]`